mipengine/algorithms/algorithm.py

```diff
@@ -140,12 +129,7 @@ def get_specification() -> AlgorithmSpecification:
         pass
 
     @abstractmethod
-    def run(self, engine):
-        # The executor must be availiable only inside run()
-        # The reasoning for this is that executor.data_model_views must already be
-        # available when the executor is available to the algorithm, but the creation of
-        # the executor.data_model_views requires calls to
-        # algorithm.get_variable_groups(), algorithm.get_check_min_rows() and get_dropna().
+    def run(self, engine, data_model_views, metadata):
```

Comment:
The algorithm will run fine even if the input data are not database views, but actual tables. Also, the "data model" concept is irrelevant here. We could call the parameter `input_data`, but of course it's "input" data and not "output" data (it's a function parameter). 

So the parameter can be just called `data` which also sits nicely with `metadata` just next to it. The name might seem very generic but it's perfectly clear what "data" is in the algorithm context.